### PR TITLE
Added 1.8 Support

### PR DESCRIPTION
**Pull request by entity**
Hi,

When using this on 1.8.x I would get the following error:
`java.lang.NoSuchMethodError: org.bukkit.scheduler.BukkitScheduler.runTaskTimer(Lorg/bukkit/plugin/Plugin;Ljava/util/function/Consumer;JJ)V`

Took this upon myself to fix, and works perfectly now :)

### CODE REVIEW

File: src/main/java/com/hazebyte/base/Base.java [hunk 1]
```diff
@@ -249,14 +250,17 @@ public boolean close(HumanEntity entity) {
...
-            } else {
-                if (!bukkitTask.isCancelled()) bukkitTask.cancel();
+
+        new BukkitRunnable() {
+            @Override
+            public void run() {
+                Inventory inventory = entity.getOpenInventory().getTopInventory();
+                if (inventory.getHolder() instanceof Base && inventory.getViewers().size() > 0) {
+                    inventory.getViewers().forEach(Base.this::update);
+                } else {
```
Comment (imWillX) [inline src/main/java/com/hazebyte/base/Base.java]: Missing ```if (!bukkitTask.isCancelled()) bukkitTask.cancel();```

**Comment by imWillX** [comment]
Thanks for the pull request. Does MC 1.8 not support java 8? If so, we happen to have consumers, lambdas, etc all over codebase.

**Comment by entity** [comment]
MC 1.8 should support it, but not sure it nulled out for me. Np though!

Also should be a fix for the ArrayIndexOutOfBoundsException when clicking outside the inventory :)

File: src/main/java/com/hazebyte/base/Base.java (hunk 1 above)
Comment (imWillX) [inline src/main/java/com/hazebyte/base/Base.java:260]:
```suggestion
                } else {
                    if (!bukkitTask.isCancelled()) bukkitTask.cancel();
```

File: src/main/java/com/hazebyte/base/Base.java [hunk 2]
```diff
@@ -356,6 +360,8 @@ public void attach(Button button) {
      * @param event
      */
     public void onInventoryClick(InventoryClickEvent event) {
+        if(event.getClickedInventory() == null) return;
```
Comment (imWillX) [inline src/main/java/com/hazebyte/base/Base.java]:
```suggestion
        if (event.getClickedInventory() == null) return;
```

File: src/main/java/com/hazebyte/base/Base.java [hunk 3]
```diff
@@ -356,6 +360,8 @@ public void attach(Button button) {
...
+        if(event.getRawSlot() < 0) return;
```
Comment (imWillX) [inline src/main/java/com/hazebyte/base/Base.java]:
```suggestion
        if (event.getRawSlot() < 0) return;
```

**Review by imWillX — CHANGES_REQUESTED** [review]

**Comment by imWillX** [comment]
Sure, looks good. Just requesting a few changes and we can merge it in.

File: src/main/java/com/hazebyte/base/Base.java [hunk 4]
```diff
@@ -356,6 +361,8 @@ public void attach(Button button) {
      * @param event
      */
     public void onInventoryClick(InventoryClickEvent event) {
+        if (event.getClickedInventory() == null) return;
+        if(event.getRawSlot() < 0) return;
```
Comment (imWillX) [inline src/main/java/com/hazebyte/base/Base.java]:
```suggestion
        if (event.getRawSlot() < 0) return;
```

**Review by imWillX — CHANGES_REQUESTED** [review]

**Review by imWillX — APPROVED** [review]

**Pull request merged**